lib/github/middleware.js

```diff
@@ -38,11 +52,21 @@ module.exports = function middleware (callback) {
   return withSentry(async (context) => {
     enhanceOctokit(context.github, context.log)
 
-    context.sentry.setExtra('GitHub Payload', {
-      event: context.name,
-      action: context.payload.action,
-      id: context.id,
-      repo: (context.payload.repository) ? context.repo() : undefined
+    if (newrelic !== undefined) {
+      newrelic.addCustomAttributes({
+        'Webhook ID': context.id,
+        'Webhook Event': `${context.name}.${context.payload.action}`,
+        'Repository': context.payload.repository
+      })
+    }
+
+    withNRSegment('middleware:sentry.setExtra', () => {
```

Comment:
I'm not sure we need to trace this, it isn't a web call, it just sets keys on the sentry scope object.

Reply:
I thought that, however if you look down we actually run `context.repo()`. So I'm wondering what the execution time would be for that. I think that's actually making an API request against GitHub.

Reply:
Looking at [the source](https://github.com/probot/probot/blob/6e45ad7/src/context.ts#L106) (note this is the newer probot source) it's just a lookup in the object. That explains why this works without an `await` too!

Reply:
Oh thanks! Good call. I'll revert this back then. I agree setting those values shouldn't need to be instrumented then.

Reply:
Found the same set in the [7.x source](https://github.com/probot/probot/blob/7.x/src/context.ts#L63-L74) that's definitely a synchronous object property lookup!